sdk/core/core-tracing/src/createSpan.ts

```diff
@@ -48,6 +64,77 @@ export function isTracingDisabled(): boolean {
   return Boolean(azureTracingDisabledValue);
 }
 
+/**
+ * Maintains backwards compatibility with the previous `OperationTracingOptions` which included `SpanOptions`
+ */
+function disambiguateParameters<T extends { tracingOptions?: OperationTracingOptions }>(
+  operationOptions: T,
+  startSpanOptions?: SpanOptions
+): [OperationTracingOptions, SpanOptions, T] {
+  const { tracingOptions } = operationOptions;
+
+  // If startSpanOptions is provided, then we are using the new signature,
+  // otherwise try to pluck it out of the tracingOptions.
+  const spanOptions: SpanOptions = startSpanOptions || (tracingOptions as any)?.spanOptions || {};
+  spanOptions.kind = spanOptions.kind || SpanKind.INTERNAL;
+
+  return [tracingOptions || {}, spanOptions, operationOptions];
+}
+
+/**
+ * Creates a new span using the given parameters.
+ *
+ * @param spanName - The name of the span to created.
+ * @param spanOptions - Initialization options that can be used to customize the created span.
+ * @param tracingContext - The tracing context to use for the created span.
+ *
+ * @returns - A new span.
+ */
+function startSpan(spanName: string, spanOptions: SpanOptions, tracingContext: Context) {
+  if (isTracingDisabled()) {
+    return trace.wrapSpanContext(INVALID_SPAN_CONTEXT);
+  }
+
+  const tracer = getTracer();
+  return tracer.startSpan(spanName, spanOptions, tracingContext);
+}
+
+/**
+ * Adds the `az.namespace` attribute on a span, the tracingContext, and the spanOptions
+ *
+ * @param span - The span to add the attribute to in place.
+ * @param tracingContext - The context bag to add the attribute to by creating a new context with the attribute.
+ * @param namespace - The value of the attribute.
+ * @param spanOptions - The spanOptions to add the attribute to (for backwards compatibility).
+ *
+ * @internal
+ *
+ * @returns The updated span options and context.
+ */
+function setNamespaceOnSpan(
+  span: Span,
+  tracingContext: Context,
+  namespace: string,
+  spanOptions: SpanOptions
+) {
+  span.setAttribute(knownSpanAttributes.AZ_NAMESPACE.spanAttributeName, namespace);
+  const updatedContext = tracingContext.setValue(
+    knownSpanAttributes.AZ_NAMESPACE.contextKey,
+    namespace
+  );
+
+  // Here for backwards compatibility, but can be removed once we no longer use `spanOptions`
+  const updatedSpanOptions = {
```

Comment:
The redundancy doesn't hurt anything today like you said because it's the same name and value, but for future my thought was to give the attribute on the context higher precedence in tracingPolicy than the one set on spanOptions (which is done as part of this PR)

This way if they drift before we get a chance to remove this it'll still do the right thing for anyone using the context bag for az.namespace